robot/client/client_test.go

```diff
@@ -726,6 +729,53 @@ func TestClientDisconnect(t *testing.T) {
 	test.That(t, err, test.ShouldBeError, client.checkConnected())
 }
 
+func TestClientDisconnectError(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	listener, err := net.Listen("tcp", "localhost:0")
+	test.That(t, err, test.ShouldBeNil)
+
+	withInterceptor := grpc.ChainUnaryInterceptor(
+		func(
+			ctx context.Context,
+			req interface{},
+			info *grpc.UnaryServerInfo,
+			handler grpc.UnaryHandler,
+		) (interface{}, error) {
+			if strings.HasSuffix(info.FullMethod, "RobotService/GetStatus") {
+				return nil, status.Errorf(codes.Unknown, readWriteOnClosedPipeErrorMsg)
+			}
+			var resp interface{}
+			return resp, nil
+		},
+	)
+	gServer := grpc.NewServer(withInterceptor)
+
+	injectRobot := &inject.Robot{}
+	injectRobot.ResourceRPCSubtypesFunc = func() []resource.RPCSubtype { return nil }
+	injectRobot.StatusFunc = func(ctx context.Context, rs []resource.Name) ([]robot.Status, error) {
+		return []robot.Status{}, nil
+	}
+	pb.RegisterRobotServiceServer(gServer, server.New(injectRobot))
+
+	go gServer.Serve(listener)
+
+	test.That(t, err, test.ShouldBeNil)
+
+	client, err := New(
+		context.Background(),
+		listener.Addr().String(),
+		logger,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	_, err = client.Status(context.Background(), []resource.Name{})
+	test.That(t, status.Code(err), test.ShouldEqual, codes.Unavailable)
```

Comment:
can you test this on unary + streaming along with verifying the fancy error string is there?